tutorials/desktop/firefox-for-web-developers/firefox-for-web-developers.md

```diff
@@ -0,0 +1,309 @@
+
+The **Toggle Tools** entry has a shortcut of **Ctrl-Shift-I**. Go on, give it a try. Each time you press it, the panel will be opened or closed, depending on its previous state. For historical reasons there's also another shortcut that will do the same thing, and which is often easier to use if you have a full-sized keyboard: the **F12** key. Pressing **F12** then selecting a particular tool with the mouse is usually faster than navigating directly to it using the **Web Developer** menu, hence why most of the entries in that menu are rarely used.
+
+
+### Using the context menu ###
+There is a third way to open the developer tools. This is not a toggle, as it always switches to a particular tab, but is a very common approach when you want to take a closer look at a specific element on the web page. If you right-click on something in the page to open the context menu, you'll find an entry labelled **Inspect Element** towards the bottom. You can select this to open the developer tools with the **Inspector** tab active and the element you clicked on already highlighted within the panel. Let's give it a try:
+
+1. Right click on the "Do more with Ubuntu!" heading near the top of the web page.
+2. Select **Inspect Element** from the context menu.
+3. The developer tools should open with the Inspector tab selected and the `<h1>` element highlighted.
+
+![Inspecting the page heading](images/inspector.png)
+
+As you can see, this method cuts right through the layers of HTML content to highlight the specific element that you right-clicked on, making this a very convenient way to open the tools if you need to work on a particular part of the page.
+
+
+## The inspector##
+Duration: 8:00
+
+All the tabs in the developer tools panel have their uses, but for beginners to web development — or anyone who is simply curious about how a web page works — you only really need to use the **Inspector** and the **Console**. This section and the next will describe them in a little more detail.
+
+The inspector lets you look at the details of the web page's "Document Object Model" (DOM). This is the browser's internal version of the page structure, and the inspector exposes it in a way that makes it look similar to the HTML that was used to create the page. Be clear, though, that although the DOM is initially created from the HTML, it's not quite the same thing. The DOM is a live representation of the page structure, so if the page content has been dynamically modified by some JavaScript code the changes will be reflected in the DOM, and hence in the inspector. The live nature of the DOM also means that you can change things in the inspector and see the results in real-time on the page — ideal for interactively tweaking the design of your page.
```

Comment:
You could mention that https://tutorials.ubuntu.com is actually a very good example of the DOM not being the same thing as the page source. Since Tutorials is actually a Polymer app, very little of the page content you see if actually in the page's source. But don't mention it if you feel it's irrelevant. It's your call.

Reply:
Not irrelevant, but perhaps getting a bit too technical for the audience this is intended for. I'm envisioning this as a tutorial for someone who is only just learning web development, or who just wants to poke around under the hood of a web page - if they know what Polymer is then they're probably not the target demographic.

Reply:
Yep, I agree.